python-sdk/src/astro/databases/base.py

```diff
@@ -183,6 +183,21 @@ def default_metadata(self) -> Metadata:
         """
         raise NotImplementedError
 
+    def openlineage_dataset_namespace(self) -> str:
+        """
+        Returns the open lineage dataset namespace as per
+        https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+        """
+        airflow_conn = self.hook.get_connection(self.conn_id)
+        return f"{self.sql_type}://{airflow_conn.host}"
```

Comment:
What are the pro/cons of using this vs using `from openlineage.common.dataset import Source`

```python
from openlineage.common.dataset import Source

airflow_conn = self.hook.get_connection(self.conn_id)
Source(scheme=airflow_conn.scheme, connection_url={airflow_conn.host})
```

cc @mobuchowski @harels @conorbev @julienledem

Reply:
We want to deprecate `openlineage.common.dataset` models and use real OpenLineage models in extractors.